### PR TITLE
feat: xgboost prediction for player movements and rotation

### PR DESCRIPTION
**Pull request by microcrit**
stems from playeraimbot branch which this feature depends on. will likely cause merge conflicts in the future.

should improve accuracy and make "prediction" for playeraimbot make sense.   
may end up making aiming logic more accurate

see [xgboost repo](https://github.com/cigaretteclient/xgboost) for trained models. not 100% sure how models will be downloaded but i'll find a way.   
if this is not needed please delete the branch and i will switch back to the aimbot branch.

### CODE REVIEW

**Comment by microcrit** [comment]
BTW the same group that made xgboost-java also made the lightgmb4j bindings, if we want memory efficiency we might want [to use it instead](https://mvnrepository.com/artifact/io.github.metarank/lightgbm4j)
